assume/common/utils.py

```diff
@@ -316,11 +320,16 @@ def aggregate_step_amount(orderbook: Orderbook, begin=None, end=None, groupby=[]
                 until=bid["end_time"],
             )
             for date in starts:
-                deltas.append((date, bid["volume"]) + add)
-                deltas.append(
-                    (date + timedelta(hours=duration_hours), -bid["volume"]) + add
+                deltas.extend(
+                    (
+                        (date, bid["volume"]) + add,
+                        (
+                            date + timedelta(hours=duration_hours),
+                            -bid["volume"],
+                        )
+                        + add,
+                    )
```

Comment:
if you want to reduce appends you should also list comprehend the for loop `for date in starts:`.

Well I don't know - thats probably not our main bottleneck is it?

Reply:
yeah, could be helpful as well. Maybe not a bottleneck, but some optimization and nices looking code is still nice.